clippy_lints/src/option_if_let_else.rs

```diff
@@ -165,6 +165,13 @@ fn try_get_option_occurrence<'tcx>(
             }
 
             let mut app = Applicability::Unspecified;
+
+            let (none_body, is_argless_call) = if let Some(call_expr) = try_get_argless_call_expr(none_body) {
+                (call_expr, true)
+            } else {
+                (none_body, false)
+            };
```

Comment:
In fact, `try_get_argless_call_expr()` is now so short and used at only one place that it can probably be inlined (and the function removed):

```suggestion
            let (none_body, is_argless_call) = match none_body.kind {
                ExprKind::Call(call_expr, []) if !none_body.span.from_expansion() => (call_expr, true),
                _ => (none_body, false),
            };
```